debian/postinst

```diff
@@ -142,23 +141,50 @@ case "$1" in
             chmod 770 /var/lib/jigasi
         fi
 
+        JIGASI_XMPPPWD=$(echo -n $JIGASI_XMPP_SECRET | base64 -)
+
         # populate the config with debconf values
-        sed -i "s/<<JIGASI_SIPUSER>>/$JIGASI_SIPUSER/g" /etc/jitsi/jigasi/sip-communicator.properties
-        sed -i "s/<<JIGASI_SIPPWD>>/$JIGASI_SIPPWD/g" /etc/jitsi/jigasi/sip-communicator.properties
+        JIGASI_CONFIG="/etc/jitsi/jigasi/sip-communicator.properties"
+        sed -i "s/<<JIGASI_SIPUSER>>/$JIGASI_SIPUSER/g" $JIGASI_CONFIG
+        sed -i "s/<<JIGASI_SIPPWD>>/$JIGASI_SIPPWD/g" $JIGASI_CONFIG
         SIPSERVER=`echo $JIGASI_SIPUSER | cut -d'@' -f 2`
-        sed -i "s/<<JIGASI_SIPSERVER>>/$SIPSERVER/g" /etc/jitsi/jigasi/sip-communicator.properties
-        sed -i "s/<<DOMAIN_BASE>>/$JVB_HOSTNAME/g" /etc/jitsi/jigasi/sip-communicator.properties
+        sed -i "s/<<JIGASI_SIPSERVER>>/$SIPSERVER/g" $JIGASI_CONFIG
+        sed -i "s/<<DOMAIN_BASE>>/$JVB_HOSTNAME/g" $JIGASI_CONFIG
+        sed -i "s/<<JIGASI_XMPPPWD>>/$JIGASI_XMPPPWD/g" $JIGASI_CONFIG
 
         PROSODY_HOST_CONFIG="/etc/prosody/conf.avail/$JVB_HOSTNAME.cfg.lua"
         if [ -f $PROSODY_HOST_CONFIG ]; then
-            JIGASI_PROSODY="Component \"callcontrol.$JVB_HOSTNAME\" component_secret = \"$JIGASI_SECRET\""
-            # update prosody config
-            if [ ! "$(grep "Component \"callcontrol.$JVB_HOSTNAME\"" $PROSODY_HOST_CONFIG)" ]; then
-                echo $JIGASI_PROSODY >> $PROSODY_HOST_CONFIG
-                invoke-rc.d prosody restart
+            USER_EXISTS_CHECK=`prosodyctl adduser jigasi@auth.$JVB_HOSTNAME < /dev/null || true`
+            if [ ! "$USER_EXISTS_CHECK" = "That user already exists" ]; then
+                prosodyctl register jigasi auth.$JVB_HOSTNAME $JIGASI_XMPP_SECRET || true
```

Comment:
This is a bit confusing. The goal is to create a new user if one doesn't exists, and otherwise do what? Keep the previous password or change it? A log message might be useful.

I looked up the difference between `adduser` and `register`, and they should accomplish the same, except `register` is maintained for compatibility with `ejabberdctl`. Can we just pass the password to `adduser` and check the return status?

Reply:
Well, register has a param with a password, adduser reads it from stdin. 
Well we create the user once and put the password in the template once, so all of the rest configures on the update just skip.

Reply:
So the first time this runs the user does not exist. We add a user via `adduser`, and then also run `register` which updates the password?

And the subsequent runs when the user exists, we do nothing?

If so, I think we can simplify to just `adduser` with the password, e.g.:
```
echo $JIGASI_XMPP_SECRET | prosodyctl adduser jigasi@auth.$JVB_HOSTNAME || true
```

Reply:
> And the subsequent runs when the user exists, we do nothing?

Yes.

> If so, I think we can simplify to just adduser with the password, e.g.:

But this will update the password on every update and will break it after the first one.

This code was borrowed from: https://github.com/jitsi/jitsi-meet/blob/29fa4c935e9449f000d23801e001663ab882ec74/debian/jitsi-meet-prosody.postinst#L131

Reply:
So `adduser` with empty input fails with `That user already exists`, but `adduser` with non-empty input updates the password?